src/script/finale.ts

```diff
@@ -4,17 +4,40 @@ import { HelpProvider } from "./help/helpProvider";
 import { Model } from "./modelParser";
 import { PathHelper } from "./pathHelper";
 import { ViewFactory } from "./viewFactory";
+import { isNullOrUndefined } from "util";
 
 document.addEventListener("DOMContentLoaded", () => {
-    generateDocument(document.location.search);
+    if (isNullOrUndefined(sessionStorage.getItem(document.location.search.slice(4)))) {
+        generateDocument(document.location.search);
+        } else {
+            parsedContent = JSON.parse(sessionStorage.getItem(document.location.search.slice(4)));
+            LoadDOM(document.location.search);
+        } 
     document.onkeyup = (e) => {
         if (InputHelper.IsHelpKeyBinding(e)) {
             HelpProvider.toggleVisibility();
         }
     };
 }, false);
 
+let parsedContent = null;
```

Comment:
You should consider passing `parsedContent` as a parameter, instead of making it a global variable.